3d_armor/init.lua

```diff
@@ -436,80 +436,64 @@ end, true)
 
 minetest.register_globalstep(function(dtime)
 	timer = timer + dtime
-	if timer > armor.config.init_delay then
-		for player, count in pairs(pending_players) do
-			local remove = init_player_armor(player) == true
-			pending_players[player] = count + 1
-			if remove == false and count > armor.config.init_times then
-				minetest.log("warning", S("3d_armor: Failed to initialize player"))
-				remove = true
-			end
-			if remove == true then
-				pending_players[player] = nil
+	if timer <= armor.config.init_delay then
+		return
+	end
+	timer = 0
+
+	for player, count in pairs(pending_players) do
+		local remove = init_player_armor(player) == true
+		pending_players[player] = count + 1
+		if remove == false and count > armor.config.init_times then
+			minetest.log("warning", S("3d_armor: Failed to initialize player"))
+			remove = true
+		end
+		if remove == true then
+			pending_players[player] = nil
+		end
+	end
+
+	-- water breathing protection
+	if armor.config.water_protect == true then
+		for _,player in pairs(minetest.get_connected_players()) do
+			local name = player:get_player_name()
+			if name and armor.def[name].water > 0 and
+					player:get_breath() < 10 then
+				player:set_breath(10)
 			end
 		end
-		timer = 0
 	end
 end)
 
 -- Fire Protection and water breathing, added by TenPlus1.
 
 if armor.config.fire_protect == true then
-	-- override hot nodes so they do not hurt player anywhere but mod
-	for _, row in pairs(armor.fire_nodes) do
-		if minetest.registered_nodes[row[1]] then
-			minetest.override_item(row[1], {damage_per_second = 0})
-		end
-	end
+	-- override torch damage
```

Comment:
IMO iterating `armor.fire_nodes` is a much saner solution. It would also allow better mod inter-compatibility by only changing `damage_per_second` to `1` if it was `0`/nil.

Reply:
Amended.